lib/inspec/profile.rb

```diff
@@ -239,6 +239,15 @@ def info(res = params.dup) # rubocop:disable Metrics/CyclomaticComplexity
         data[:impact] = 1.0 if data[:impact] > 1.0
         data[:impact] = 0.0 if data[:impact] < 0.0
         data[:id] = id
+
+        # if the code field is empty try and pull info from dependencies
+        if data[:code].empty? && parent_profile.nil?
+          locked_dependencies.dep_list.each do |_name, dep|
```

Comment:
Could this be extracted to a method to avoid `Metrics/PerceivedComplexity`?

Reply:
Tried that, even putting a method call to a new one will blow it. We will need to redo this method in full to break it up more. Due to the nature of this bandaid hot fix we will put it under techDebt.

Reply:
Understood. No judgment here!